configuration/file-configuration.ts

```diff
@@ -0,0 +1,56 @@
+import {readFile, accessSync, R_OK} from "fs";
+import {IConfiguration} from "./iconfiguration";
+
+export class FileConfiguration implements IConfiguration {
+    private userConfig: { [key: string]: any } = {};
+
+    /**
+     * Asynchronously initialize configuration values from the passed file.
+     *
+     * @param {string} configFilename - Name of the JSON file containing
+     * configuration preferences.
+     */
+    public async initialize(configFilename: string): Promise<void> {
+        try {
+            accessSync(configFilename, R_OK);
```

Comment:
apparently fs.exists is deprecated now. Suggest using the async version of access though. 
Also, we should just check for F_OK access: if the file exists but we can't read it, we should just throw.  Silently swallowing the error would not be intuitive to the user.